### ISSUE TITLE
Whole project visibility

### ISSUE THREAD

**Issue by MasterBowler** [microsoft/vscode-copilot-release#562]
I would like for github copilot to be able to read and understand every file in the project.

Diagram/image/pdf generation would also be nice.

**Comment by MarvinButh** [comment]
Good point. I think when copilot finally migrates to GPT-4-Turbo, this will be possible. With just 16k tokens you can't really provide project context understanding, but with the new 120k tokens it probably at least be able to use the last few opened files or the files that are in the same directory/references to other files as context.

**Comment by voltussprint** [comment]
Forget project context understanding.  I am on a certain file which I just generated, and asked for changes to that file, it is not able to do it.

**Comment by skoonin** [comment]
yea would be great if you can tell copilot which files in your workspace you want it to consider in a single context too! (in case I have folders that are not related to a certain project in the same workspace)

**Comment by ketpain** [comment]
100% needed to make complete use of the AI. Even WITHOUT the 120k token limit, if the AI could just go into source files and get a basic idea or run down of what the code does a level or two down, it could give much better advice or code suggestions instead of making guesses.

**Comment by roblourens** [comment]
Not clear whether you have used `@workspace` and are asking for that or improvements to it

**Comment by MasterBowler** [comment]
> Not clear whether you have used `@workspace` and are asking for that or improvements to it


I think I tried using @workspace before and it refused reading a .txt file containing info about the project.
I tried it again and with @workspace I am indeed able to make copilot read the whole code base.
Thanks, I will close the issue.
